x-pack/plugins/security_solution/public/common/containers/source/use_data_view.tsx

```diff
@@ -25,26 +27,71 @@ import {
   isErrorResponse,
 } from '../../../../../../../src/plugins/data/common';
 import * as i18n from './translations';
-import { getBrowserFields, getDocValueFields } from './';
 import { SourcererScopeName } from '../../store/sourcerer/model';
 import { getSourcererDataview } from '../sourcerer/api';
 
-const getEsFields = memoizeOne(
-  (fields: IndexField[]): FieldSpec[] =>
-    fields && fields.length > 0
-      ? fields.map((field) =>
+export type IndexFieldSearch = (param: {
+  dataViewId: string;
+  scopeId?: SourcererScopeName;
+  needToBeInit?: boolean;
+  cleanCache?: boolean;
+}) => Promise<void>;
+
+type DangerCastForBrowserFieldsMutation = Record<
+  string,
+  Omit<BrowserField, 'fields'> & { fields: Record<string, BrowserField> }
+>;
+interface DataViewInfo {
+  browserFields: DangerCastForBrowserFieldsMutation;
+  docValueFields: DocValueFields[];
+  indexFields: FieldSpec[];
+}
+
+/**
+ * HOT Code path where the fields can be 16087 in length or larger. This is
+ * VERY mutatious on purpose to improve the performance of the transform.
+ */
+const getDataViewStateFromIndexFields = memoizeOne(
```

Comment:
This function has been created for efficiency purposes, it processes `browserFields`, `indexFields` and `docValueFields` all at once, previously it was done in 3 different functions (`getBrowserFields`, `getDocValueFields`, `getIndexFields`), each one was iterating all items.

Reply:
Yeah, you are right, I had the same thought, the old functions are still called in the same place they're defined (_common/containers/source/index.tsx_) by the `useFetchIndex` hook. 
But the problem is that the structure it needs in the state for the `indexPatterns` (`getIndexFields`) is a little bit different than the DataView (we had our own `getEsFields` before), we would need to pass a flag parameter to the `getDataViewStateFromIndexFields` (better not), or implement a separate `getFetchIndexStateFromIndexFields` function to be called there.
I decided to not do it and leave it as it is because the `useFetchIndex` hook is used in many other places, but not by us, so thought that doing this would imply quite a lot of changes out of the task scope.

I will revisit the tests to see if I can improve them.